addons/mass_mailing/wizard/mailing_mailing_test.py

```diff
@@ -18,9 +18,18 @@ def send_mail_test(self):
         ctx.pop('default_state', None)
         self = self.with_context(ctx)
 
+        notification_messages = []
         mails_sudo = self.env['mail.mail'].sudo()
+        test_emails = []
+        for candidate in self.email_to.splitlines():
+            test_email = tools.email_split(candidate)
```

Comment:
We could build two list, valid_emails and invalid_candidates. Then build a message with "Mailing adresses incorrect: join-all-invalid-candidates". That way we don't do a list of message but manage lists of emails. Then we build the final message to post.